exp/api/v1beta1/azuremanagedcontrolplane_types.go

```diff
@@ -247,6 +251,122 @@ type AzureManagedControlPlaneStatus struct {
 	LongRunningOperationStates infrav1.Futures `json:"longRunningOperationStates,omitempty"`
 }
 
+// AutoScalerProfile parameters to be applied to the cluster-autoscaler.
+// See the [FAQ](https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/FAQ.md#what-are-the-parameters-to-ca) for more details about each parameter.
+type AutoScalerProfile struct {
+	// BalanceSimilarNodeGroups - Valid values are 'true' and 'false'. The default is false.
+	// +kubebuilder:validation:Enum="true";"false"
+	// +optional
+	BalanceSimilarNodeGroups *BalanceSimilarNodeGroups `json:"balanceSimilarNodeGroups,omitempty"`
```

Comment:
@CecileRobertMichon @nojnhuh while reviewing this it occurs to me that this approach is different than the one I took in this PR:

- https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/2781

An example:

- https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/2781/files#diff-585942e07c6d3dd4ca6c2d35efdca0fa27c6a7bf6e3d6711648edfad330fb3adR46-R49

I think the approach here is safer, but perhaps requires more maintenance. I'm happy to adopt it in my PR, I'd love to have these type definitions be standardized.

Reply:
I'm surprised it's different, I thought this approach above was exactly what we talked about doing for you PR as well when you were screen sharing the other day, it's also what we're doing already in the repo in most places AFAIK, eg. https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/api/v1beta1/types.go#L459

Reply:
The approach I’m talking about is wrapping the string type in a strongly typed “string” like thing, and then having a const for each allowed enum value.

I’m pretty sure we’re not doing this everywhere, but let’s do this from here on out. The only downside I can see to this is more boilerplate overhead; I can’t see a reason (other than dev laziness) not to do it this way as a rule. Curious if there are other reasons mot to bother creating our own types + consts.

Reply:
Since it's been done both ways in the code I'm glad I chose wisely... While the additional boilerplate was kind of annoying, especially with the pointer aspect, it was pretty minor and a one-time thing. I do like it better than just calling it a string in the type.

Reply:
I'll adapt my PR to follow this pattern as well.